core/services/workflows/engine.go

```diff
@@ -122,7 +123,7 @@ type Engine struct {
 	workflow             *workflow
 	secretsFetcher       SecretsFor
 	env                  exec.Env
-	localNode            capabilities.Node
+	localNode            atomic.Pointer[capabilities.Node]
```

Comment:
Weird that this is needed... localNode is set only in init() and init() doesn't exit until it's set (or failed). So we won't register triggers before that happens.

Reply:
Init happens in a goroutine, so if an engine is closed shortly after it's started it can race against the Close call